apps/prairielearn/src/ee/pages/instructorAiGenerateQuestion/instructorAiGenerateQuestion.ts

```diff
@@ -81,8 +94,43 @@ router.get(
   '/',
   asyncHandler(async (req, res) => {
     assertCanCreateQuestion(res.locals);
+    if (req.query?.qid) {
```

Comment:
this might be a slightly weird decision but the questionEditor page and generateNewQuestion page are the same page, where the questionEditor page is simply when a valid qid is expressed in query-parameters. That way, if a user requests a question edit page with an invalid qid, we simply fall back to the new question generation page. we could just switch to making these separate pages and simply having a redirect elsewhere.

Reply:
I think separate pages with a redirect would be much better. This would also allow us to embed the question ID / QID / whatever in the URL path (`.../:id`) rather than sticking it in the query string, which is unconventional for PL.

Reply:
We're now using separate pages.